libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -252,8 +277,34 @@ void AP_Baro::update_calibration()
     // always update the guessed ground temp
     _guessed_ground_temperature = get_external_temperature();
 
+    update_gps_calibration();
+    
     // force EAS2TAS to recalculate
     _EAS2TAS = 0;
+
+    // reset alt offset on calibration
+    _alt_offset.set_and_save(0);
+}
+
+// calibrate reference GPS altitude
+void AP_Baro::update_gps_calibration()
+{
+    float vacc;
+    const bool gps_vacc_ret = AP_GPS::gps().vertical_accuracy(vacc);
+    const uint32_t last_fix = AP_GPS::gps().last_fix_time_ms();
+    const uint32_t now = AP_HAL::millis();
+
+    // Make sure the vertical accuracy is within limits
+    if (gps_vacc_ret && vacc < BARO_MAX_GPS_ACCURACY && (now - last_fix) < BARO_MAX_GPS_DELAY) {
```

Comment:
Time handling is unneeded. Asking AP_GPS if it is healthy will apply a stricter timing check, as well as check the GPS driver itself if the GPS is reporting failure conditions.

Reply:
The PR is soon will be a year long :)

Reply:
Yeah I realize it, that's why I was pointing out that since this was opened better features became available :)

Reply:
Totally agree, will implement.